### ISSUE TITLE
ALLOWED_METHODS doesn't match regex [BUG]

### ISSUE THREAD

**Issue by MArpogaus** [bunkerity/bunkerweb#504]
**Description**
My nextcloud instance requires:

```bash
ALLOWED_METHODS=GET|POST|HEAD|COPY|DELETE|LOCK|MKCOL|MOVE|PROPFIND|PROPPATCH|PUT|UNLOCK|OPTIONS|REPORT
```
However, the regexp defined [here](https://github.com/bunkerity/bunkerweb/blob/a7535c300a6e64b8bdf90be5cfb2770858cff6df/src/common/core/misc/plugin.json#L41) dos not match for this definition.

- Is this behavior intended?
- Is there a way to disable the regex check or adapt the regexp?

**How to reproduce**
The relevant code can be found [here](https://github.com/bunkerity/bunkerweb/blob/a7535c300a6e64b8bdf90be5cfb2770858cff6df/src/common/gen/Configurator.py#L233).

```python
>>> from re import search as re_search
>>> regexp="^(?!\\|)(\\|?(GET|HEAD|POST|PUT|DELETE|CONNECT|OPTIONS|TRACE|PATCH|COPY|LOCK|MKCOL|MOVE|PROPFIND|PROPPATCH|UNLOCK)(?!.*\\2))+$"
>>> value="GET|POST|HEAD|COPY|DELETE|LOCK|MKCOL|MOVE|PROPFIND|PROPPATCH|PUT|UNLOCK|OPTIONS|REPORT"
>>> print(re_search(regexp, value))
None
```

This obviously cant work because `REPORT` is not included in the regex.

More surprisingly also the given value in the [example](https://github.com/bunkerity/bunkerweb/blob/a7535c300a6e64b8bdf90be5cfb2770858cff6df/examples/nextcloud/docker-compose.yml#L17) dos not work:

```python
>>> value="GET|POST|HEAD|COPY|DELETE|LOCK|MKCOL|MOVE|PROPFIND|PROPPATCH|PUT|UNLOCK|OPTIONS"
>>> print(re_search(regexp, value))
None
```

I found that `UNLOCK` seams to be the problematic string here.
If i remove it the regexp matches:

```python
>>> value="GET|POST|HEAD|COPY|DELETE|LOCK|MKCOL|MOVE|PROPFIND|PROPPATCH|PUT|OPTIONS"
>>> print(re_search(regexp, value))
<re.Match object; span=(0, 72), match='GET|POST|HEAD|COPY|DELETE|LOCK|MKCOL|MOVE|PROPFIN>
```

**Logs**
```log
[2023-05-25 15:04:42] - GENERATOR - ⚠️  - Ignoring variable ALLOWED_METHODS : value GET|POST|HEAD|COPY|DELETE|LOCK|MKCOL|MOVE|PROPFIND|PROPPATCH|PUT|UNLOCK|OPTIONS|REPORT doesn't match regex ^(?!\|)(\|?(GET|HEAD|POST|PUT|DELETE|CONNECT|OPTIONS|TRACE|PATCH|COPY|LOCK|MKCOL|MOVE|PROPFIND|PROPPATCH|UNLOCK)(?!.*\2))+$
```

**Possible Solution**
From my point of view, this default regexp is was to restrictive. and dos not allow to use additional non pre defined methods.
Why not use something like that:

```regexp
^(?!\\|)(\\|?[A-Z]{3,})+$
```

**Comment by TheophileDiot** [comment]
Hi @MArpogaus, indeed you're right, a mistake has been made when creating the regex. The reason why the regex is so strict is that so you know directly if you put a wrong method.
Here's the fixed version : https://regex101.com/r/RJuxQJ/1
It will be modified in the next version, thank you for opening this issue !

**Comment by fl0ppy-d1sk** [comment]
Hello @MArpogaus,

It should be fixed in the last 1.5.1 release. Thanks for your report.
